### ISSUE TITLE
CKEditor - linkit_media_library does not work for media image linking

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#7334]
# Problem
I think this is new functionality that we forgot to account for as we had a workaround before: https://sitenow.uiowa.edu/documentation/media/images-and-files#link-image 

User can place a media image and wrap it with a link using new ckeditor5 functionality. the linkit_media_library button shows on this balloon but doesn't properly link the item. The link goes to the media entity node (authenticated only) instead of changing to the absolute file path on render.

<img width="789" alt="Screenshot 2024-02-05 at 11 38 26 AM" src="https://github.com/uiowa/uiowa/assets/4663676/80afe4ff-fac4-464b-9272-75558ee4a402">
<img width="871" alt="Screenshot 2024-02-05 at 11 38 46 AM" src="https://github.com/uiowa/uiowa/assets/4663676/1cd7ddbc-9580-4866-abdb-1322355c44a5">

# Known Issue

https://sitenow.uiowa.edu/known-issues#uiowa-issue-7334

# Steps to reproduce

Add an image to the WYSIWYG and click the link option, select a file to link to.

# Proposed solution

Either scope to not show the linkit_media_library button here or figure out what needs to change for it to work?

**Comment by briand44** [comment]
Grooming conversation... May want to make community aware of this while we wait to plan this. Joe is volunteering.

**Comment by joewhitsitt** [comment]
Made a comment on the community issue: https://www.drupal.org/project/linkit_media_library/issues/3329877#comment-15432194

**Comment by briand44** [comment]
Blocked pending our investigation https://github.com/orgs/uiowa/projects/20/views/1?pane=issue&itemId=52575565

**Comment by pyrello** [comment]
Initial experiment proposal: Place `console.log()` statements in the `media_linkit_library` CKE5 plugin to try to determine the point where the expected attributes are not being added to the link when it is inserted into the textarea.

**Comment by joewhitsitt** [comment]
Marking as upstream because of https://github.com/uiowa/uiowa/issues/7348#issuecomment-1936273238

**Comment by joewhitsitt** [comment]
Interesting comment here https://www.drupal.org/project/linkit/issues/3396049#comment-15450593 which I interpret as, Core needs to solve this and _will_ by basically building the majority of Linkit into core. Only after that is done can Linkit possibly fix as well.

**Comment by briand44** [comment]
Postponed until progress upstream
